### PR TITLE
[DNM] Buff and Refactor the Foodcart

### PR DESCRIPTION
**Pull request by PacifistDalek**
## What Does This PR Do

**UPDATED**

Buffs the food cart from having 6 slots for food and 6 slots for drinks to ~~15 and 15.~~ like 1500 each. 

Refactored the food cart to fix bad code.

## Why It's Good For The Game
The food cart is underutilized by chefs, and giving it more space to hold culinary masterpieces and glasses of bilk should allow for more uses as an around-the-station food delivery service, with less trips back to the kitchen and bar to restock.

Plus, we already live on a station with trays that can hold crazy amounts of food, shouldn't an entire cart be able to do the same?

## Changelog
:cl:
tweak: buffed the food cart's storage capacity
/:cl:

### CODE REVIEW

**Comment by Mitchs98** [comment]
I'd argue to raise it even higher to 30 and 30, but this is definitely a nice buff.

**Review by SteelSlayer — COMMENTED** [review]
The current way the food cart handles items is actually really awful. I don't blame you for just copying how the current code is but I'd ask that you not perpetuate it by adding more hardcoded slots like this. Instead try to make it more expandable and only require a few lines.

A small example of what I mean:
```dm
for(var/obj/item/I in food_items)
	dat += "<a href='?src=[UID()];food=[I]'>[I]</a><br>"
for(var/obj/item/I in drink_items)
	dat += "<a href='?src=[UID()];drink=[I]'>[I]</a><br>" 
```
This condenses everything from line 78 to 107 to just these 4 lines. Obviously you need some more code elsewhere for it to work fully, but I hope that shows what you could do with it. If you decide to refactor this and need help, I can probably help out.

File: code/game/objects/structures/foodcart.dm [hunk 1]
```diff
@@ -139,6 +175,51 @@
 			user.put_in_hands(food_slots[6])
 			to_chat(user, "<span class='notice'>You take [food_slots[6]] from [src].</span>")
 			food_slots[6] = null
+	if(href_list["f7"])
```
Comment (AffectedArc07) [inline code/game/objects/structures/foodcart.dm:178]: Absolutely fucking not. Refactor this to make it use one single eject href, or TGUI it. Repetition like this is not happening.

**Review by AffectedArc07 — CHANGES_REQUESTED** [review]

**Comment by Carthusia** [comment]
For chefs like me, even 30 is a bit low, as that would be only 10 plates of 3 different meals.

It is a giant car designed specifically to hold food, currently loading food into a duffle bag would get more than 15

**Comment by Mitchs98** [comment]
> For chefs like me, even 30 is a bit low, as that would be only 10 plates of 3 different meals.
> 
> It is a giant car designed specifically to hold food, currently loading food into a duffle bag would get more than 15

Indeedle. But 30 would be a nice buff.
Honestly I'd say even 50 would be good. It's not like it's a super overpowered item, all you do is store food in it.

**Review by variableundefined — COMMENTED** [review]
Objection on code ground. Approval on design ground. 

Please remedy the code issue within 14 days or it will be closed.

**Review by Kyep — APPROVED** [review]

**Comment by PacifistDalek** [comment]
I'm definitely going to refactor with some help from the discord coding chat, and I'll buff it further than 15!

**Comment by PacifistDalek** [comment]
The food cart is now refactored into a smartfridge variant. This completely purges the awful code, buffs the food cart to have functionally UNLIMITED storage, and gives it a nice TGUI all in one go.

Only real issue here is that it's going to require a map edit to go through.

Marked as Do Not Merge because I still need to put in the map edit, and I need to make it either have a circuit board so that it can be correctly deconstructed and constructed, or make it impossible to deconstruct (i.e. the way it worked pre refactor). Same for upgrading it too.

**Comment by AffectedArc07** [comment]
```
loading ./map_files/MetaStation/MetaStation.v41A.II.dmm
_maps/map_files/MetaStation/MetaStation.v41A.II.dmm:54984:error: undefined type: /obj/structure/foodcart
_maps/map_files/MetaStation/MetaStation.v41A.II.dmm:54988:error: unknown type
loading ./map_files/Delta/delta.dmm
_maps/map_files/Delta/delta.dmm:32979:error: undefined type: /obj/structure/foodcart
_maps/map_files/Delta/delta.dmm:32982:error: unknown type

```

**Comment by variableundefined** [comment]
I'm going to put [DNM] as a prefix for clarity. I'm also putting on map edit tags because this PR will eventually require a map edit and it help us understand why it is blocked.

**Comment by PacifistDalek** [comment]
Thank you, sounds perfect to me. This is a lower priority for me until the map freeze is done, or at least until that gets close.

**Comment by AffectedArc07** [comment]
This can now be merged when you deconflict and update it.

If you want to remake the PR that also works.

**Comment by AffectedArc07** [comment]
PR author has not connected to server since February this year
image

This PR no longer leads mapping queue

**Review by dearmochi — APPROVED** [review]

**Comment by PacifistDalek** [comment]
> PR author has not connected to server since February this year
> image
> 
> This PR no longer leads mapping queue

No problem. I do intend to return and to work on this in the coming week, but if it's simpler and less clutter to close it, I can make a new PR when I do so. Not sure what the normal courtesy is about keeping a PR sitting open versus closing and opening a new one!

**Comment by variableundefined** [comment]
The PR has been stale for too long. Closing. Please make a new one when ready.

**Pull request closed without merging**